src/Node.php

```diff
@@ -347,22 +347,21 @@ public function getNodeKindName() : string {
      */
     public function getEndPosition() {
         // TODO test invariant - start of next node is end of previous node
-        if (isset($this->parent)) {
-            $parent = $this->parent;
-            $siblings = $parent->getChildNodes();
-            foreach ($siblings as $idx=>$nextSibling) {
-                if (spl_object_hash($nextSibling) === spl_object_hash($this)) {
-                    $siblings->next();
-                    $nextSibling = $siblings->current();
-                    return $nextSibling !== null
-                        ? $nextSibling->getFullStart()
-                        : $this->getRoot()->endOfFileToken->fullStart;
-                }
+        if (!isset($this->parent)) {
+            throw new \Exception("Unhandled node: " );
```

Comment:
not related to your change in particular, but while you're here, might as well clean up the exception message to be more informative (e.g. perhaps include the node name after the colon?)